### ISSUE TITLE
Bug: Command-line interface generator exits with success code 0 even when you pass random jibberish in the arguments

### ISSUE THREAD

**Issue by unclechu** [rakudo/rakudo#5514]
## The Problem

Take for instance `test.raku`:

``` raku
#! /usr/bin/env raku
use v6.d;

multi MAIN('foo') { say 'foo subcommand handler' }
multi MAIN('bar') { say 'bar subcommand handler' }

sub USAGE {
  print $*USAGE ~ "\n\n" ~ qq:to/USAGE/;
  Some example text
  USAGE
}
```

When I call `./test.raku --help` I get the usage info and the exit code is success (0) which is expected, as I correctly and explicitly requested usage info by `--help`. But when I call `./test.raku random-text` or just `./test.raku` I get the same usage info but it exits with success (0) while it should be error non-zero exit code it’s arguments parsing failure.

## Expected Behavior

When calling `./foo.raku` or `./foo.raku random-jibberish` to get a non-zero (error) exit code.

## Actual Behavior

I get successful exit code (0).

## Steps to Reproduce

1. Save the example from “The Problem” section above to `test.raku` file and `chmod +x test.raku`
2. Run this:

   ``` console
   $ ./foo.raku
   Usage:
     ./foo.raku foo
     ./foo.raku bar
   
   Some example text
   $ echo $?
   0
   ```
   
   ``` console
   $ ./foo.raku any-random-text
   Usage:
     ./foo.raku foo
     ./foo.raku bar
   
   Some example text
   $ echo $?
   0
   ```

## Environment
* Operating system: NixOS 23.11.2936.428544ae95ee (Tapir)
* Compiler version (`rakudo -v` or `raku -v`):
   ``` console
   $ rakudo -v
   Welcome to Rakudo™ v2023.08.
   Implementing the Raku® Programming Language v6.d.
   Built on MoarVM version 2023.09.
   ```

**Comment by ab5tract** [comment]
`USAGE` is the older way to define things. In the code there is a comment mentioning that a deprecated message should be shown but it looks like this has not yet been added.

The newer approach is [`GENERATE-USAGE(&main, |capture) {...}`](https://docs.raku.org/syntax/GENERATE-USAGE), which is more flexible but also more complicated. However, it will print to `STDERR` and exit with status 2 if triggered without `--help`.

Since `USAGE` leaves the user in charge of printing -- and doesn't provide access to passed arguments -- there is no way to conditionally divert the erroneous cases to `STDERR`.

However, at first glance it does seem trivial to at least conditionally set the exit status for `USAGE`, so I've submitted a PR to do just that.

**Comment by ab5tract** [comment]
@unclechu This should be fixed as of [a4c7d5d](https://github.com/rakudo/rakudo/commit/a4c7d5d1939f2495a0003326880199d885d42e23). Happy hacking!
